src/bfast.ts

```diff
@@ -235,67 +268,68 @@ export function parseName(name: string): [number, ArrayConstructor]{
     * Returns a number array from the buffer associated with name
     * @param name buffer name
     */
-   async getArray (name: string) {
+   async getArray (name: string): Promise<NumericArray | undefined> {
      const buffer = await this.getBuffer(name)
-     if (!buffer) return
+     if (!buffer) return undefined
      const type = name.split(':')[0]
      const Ctor = typeConstructor(type)
      const array = new Ctor(buffer)
-     return Array.from(array)
+     return array
    }
- 
+
    /**
     * Returns a single value from given buffer name
     * @param name buffer name
     * @param index row index
     */
-   async getValue (name: string, index: number) {
+   async getValue (name: string, index: number): Promise<number | BigInt | undefined> {
     const array = await this.getValues(name, index, 1)
     return array?.[0]
    }
-
+ 
    async getRange(name:string){
     const ranges = await this.getRanges()
     return ranges.get(name)
    }
-
+ 
    /**
     * Returns count subsequent values from given buffer name.
     * @param name buffer name
     * @param index row index
     * @param count count of values to return
     */
-   async getValues (name: string, index: number, count: number) {
-    if(index < 0 || count < 1) return
+   async getValues (name: string, index: number, count: number): Promise<NumericArray | undefined> {
+    if (index < 0 || count < 1) return undefined
+ 
     const range = await this.getRange(name)
-    if (!range) return
-
+    if (!range) return undefined
+ 
     const [size, ctor] = parseName(name)
-    if(size < 0) return
-
+    if(size < 0) return undefined
+ 
     const start = Math.min(range.start + index * size, range.end)
     const end = Math.min(start + size * count, range.end)
     
     const dataRange = new Range(start, end)
-    if(dataRange.length <= 0) return
-
+    if(dataRange.length <= 0) return undefined
+ 
     const buffer = await this.request(
       dataRange,
       `${name}[${index.toString()}]`
     )
-    if (!buffer) return
+    if (!buffer) return undefined
+ 
     const array = new ctor(buffer)
-
     return array
   }
   /**
    
     * Returns the buffer with given name as a byte array
     * @param name buffer name
     */
-   async getBytes (name: string) {
+   async getBytes (name: string): Promise<Uint8Array | undefined> {
      const buffer = await this.getBuffer(name)
-     if (!buffer) return
+     if (!buffer) return undefined
      const array = new Uint8Array(buffer)
```

Comment:
@vim-sroberge should this actually return an Int8Array? I believe the bytes we serialized are signed values